core/shared/src/main/scala/zio/stm/STM.scala

```diff
@@ -561,7 +570,7 @@ object STM {
 
     private[this] val txnCounter: AtomicLong = new AtomicLong()
 
-    final val globalLock = new java.util.concurrent.Semaphore(1)
+    final val globalLock = new zio.internal.LockedRef(1)
```

Comment:
This is not even using the `Int` passed into `LockedRef`, which means `globalLock` is unnecessary. You may as well just synchronize on `STM` object, e.g. `STM.synchronized { ... }` or maybe make a private object, e.g. `final val globalLock = new AnyRef { }`.

Reply:
You're right! This `LockedRef` instance does not use the passed `Int` at all.
but my understanding is that we can't use java's built-in `synchronized` in scala.js.
So, I have to come up with some data structure similar to `LockedRef` which does not take an argument.